### PR TITLE
refactor(backup): delete what the v2 migration left unreachable

### PR DESCRIPTION
**Pull request by DeJeune**
> Stacked on #17868 → #17820 → #17818 → #17817 → #17499. Review the diff against `feat/backup-auto-sync-jobs`.

### What this PR does

Before this PR:

The backup migration in the PRs below this one moved every cloud destination into `BackupService` and every scheduled backup into JobManager, but left the code they displaced sitting in place.

After this PR:

Deletion only — **711 lines out, 34 in**. Four things with no remaining way to be reached:

- **The directory-copy machinery** (`copyDirWithProgress`, `getDirSize`, symlink resolution, the copy progress handler) served the v1 full backup that #17499 deleted. Its only surviving caller — the dormant LanTransfer archive — asks for a lightweight archive and passes `skipBackupFile: true`, so none of it ran. The parameter goes with it, since only one value was ever possible.
- **`IpcChannel.BackupProgress`** had a sender and no listener anywhere: the progress dialog reads the v2 `backup.progress` IpcApi event instead.
- **`getFileContents` on both storage clients, and S3's buffer `putFileContents`** — replaced by the streaming pair in #17499 and left behind.
- **`@shared/types/backup`** stopped being cross-process when credentials stopped crossing IPC. Each client now owns the config it needs (six fields, not eleven), with no scheduling or rotation policy mixed into a storage type.

`LegacyBackupManager` goes from 537 lines to 171, and legacy backup IPC channels from 18 to 2.

Fixes: N/A

### Why we need it and why it was done in this way

Every item was verified unreachable before deletion rather than assumed: each has zero callers outside the file, and the copy machinery is reachable only through a branch whose sole live caller hardcodes the other value.

The following tradeoffs were made:

- **LanTransfer stays.** `2026-07-13-export-to-phone-temporarily-removed.md` states the service and its channels are kept dormant for a relaunch, and its 20 path-traversal tests guard code that still exists. Deleting it would overturn a documented product decision, which is not this PR's call.
- **The symlink-handling tests go with the code they test.** Losing security tests is worth naming, but they covered a copy path that no longer exists; keeping them would mean keeping 250 lines alive to be tested.
- **Config types went to each client rather than to a shared main-side module.** `WebDav` and `S3Storage` need different fields, and the merged type was how scheduling policy (`autoSync`, `syncInterval`, `maxBackups`) ended up inside a storage client's constructor argument.

The following alternatives were considered:

- **Move `WebDav.ts` / `S3Storage.ts` into `backup/destinations/`.** They are now used only from there, so it would be tidier — deferred as unrelated churn in a deletion-only PR.

Links to places where the discussion took place: N/A

### Breaking changes

None. Nothing removed here was reachable from the app.

One note recorded rather than fixed: LanTransfer's renderer-side payload producer was a `{ version: 5, localStorage }` dump — a v1 shape that in v2 carries an emoji-picker history and a favicon cache, and nothing of the user's data. It was removed with the rest of the renderer's backup service in #17868. The relaunch needs a new payload (the v2 archive), not a restoration of the old one; this is written into the entry that took the feature offline.

### Special notes for your reviewer

Worth checking that I did not delete something you consider load-bearing: the reachability argument for the copy machinery rests on `createLanTransferBackup` being its only caller and passing `skipBackupFile: true`.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Branch: This PR targets the correct branch — `main` for active development, `v1` for v1 maintenance fixes
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Documentation: A [user-guide update](https://docs.cherry-ai.com) was considered and is present (link) or not required. Check this only when the PR introduces or changes a user-facing feature or behavior.
- [x] Self-review: I have reviewed my own code (e.g., via [`/gh-pr-review`](/.claude/skills/gh-pr-review/SKILL.md), `gh pr diff`, or GitHub UI) before requesting review from others

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by xxiaoxiong — APPROVED** [review]

File: src/main/services/lanTransfer/LanTransferService.ts [hunk 1]
```diff
@@ -742,3 +744,34 @@ export class LanTransferService extends BaseService {
...
+      await fsExtra.remove(resolvedPath)
+      logger.info(`Deleted temp transfer file: ${resolvedPath}`)
+      return true
+    }
+    return false
+  } catch (error) {
+    logger.error('Failed to delete temp transfer file:', error as Error)
+    return false
+  }
+}
```
[the remark this answers is not in the document]
Comment (DeJeune) [reply src/main/services/lanTransfer/LanTransferService.ts]: Covered by `test(lan-transfer): cover deleteTransferFile's path confinement` — `LanTransferService.test.ts` has the traversal, absolute-path, prefix (`lan-transfer-evil`) and edge cases against the new home.

File: src/main/services/lanTransfer/index.ts [hunk 2]
```diff
@@ -17,5 +17,5 @@
  * └──────────┴──────────┴──────────┴───────────────┴──────────────┴────────────┴───────────┘
  */
 
-export { HANDSHAKE_PROTOCOL_VERSION, LanTransferService } from './LanTransferService'
+export { deleteTransferFile, HANDSHAKE_PROTOCOL_VERSION, LanTransferService } from './LanTransferService'
```
[the remark this answers is not in the document]
Comment (DeJeune) [reply src/main/services/lanTransfer/index.ts]: Covered by `test(lan-transfer): cover deleteTransferFile's path confinement` (see the thread above).

File: src/main/ipc.ts [hunk 3]
```diff
@@ -26,8 +26,6 @@ import { decompress } from './utils/zip'
 const logger = loggerService.withContext('IPC')
 
 export async function registerIpc() {
```
[the remark this answers is not in the document]
Comment (DeJeune) [reply src/main/ipc.ts]: Acknowledged, leaving this open for a decision: v2 creates its own staging at 0700 so the permission half of the old sweep is covered, but nothing removes temp files a pre-v2 flow left behind.
